netcheck/netcheck.go

```diff
@@ -454,11 +458,63 @@ func (c *Client) GetReport(ctx context.Context) (*Report, error) {
 		}
 	}
 
-	// TODO: if UDP is blocked, try to measure TCP connect times
-	// to DERP nodes instead? So UDP-blocked users still get a
-	// decent DERP node, rather than being randomly assigned to
-	// the other side of the planet? Or try ICMP? (likely also
-	// blocked?)
+	// Try HTTPS if UDP latency check failed for every stun server
+	// TODO: It does not show that UDP latency check failed
+	need := stuns4
+	if pc6 != nil {
+		need = append(need, stuns6...)
```

Comment:
actually, we don't need to check 6 separately. Just stuns4 alone is fine. (the derphttp connections are always IPv4 today anyway)

Reply:
Actually, we only want to do HTTP fallback if **all** UDP is blocked. Because the netcheck policy above doesn't guarantee it'll hit every server. We just relaxed it (https://github.com/tailscale/tailscale/commit/44b07aa70889d14949ad08d7ce4455207236423c) and will probably relax it more in the future.

So let's just do something like:

```go
    if !anyV4() {
         var wg sync.WaitGroup // for HTTPS checks
         for _, server := range stuns4 {
               ...
         }
         wg.Wait()
    }
```